fix(ci): Patrol E2E — androidx.test:runner 1.6.2 vs 1.5.1 버전 충돌로 Gradle build 실패

## 현상

Patrol E2E workflow 실행 시 Gradle build 에서 \`androidx.test:runner\` 버전 충돌로 **빌드 실패**. Emulator 부팅은 성공하지만 \`:app:checkDevDebugAndroidTestAarMetadata\` task 단계에서 의존성 해결 실패.

### 실패 로그

**Run**: https://github.com/Mark-Yun/minglit/actions/runs/24839817598 (2026-04-23 14:04 UTC, workflow_dispatch)

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:checkDevDebugAndroidTestAarMetadata'.
> Could not resolve all files for configuration ':app:devDebugAndroidTestRuntimeClasspath'.
   > Could not resolve androidx.test:runner:1.6.2.
     Required by:
         project :app
      > Cannot find a version of 'androidx.test:runner' that satisfies the version constraints:
           Dependency path 'app_user_android:app:unspecified' --> 'androidx.test:runner:1.6.2'
           Constraint path 'app_user_android:app:unspecified' --> 'androidx.test:runner:{strictly 1.5.1}'
             because of the following reason: version resolved in configuration ':app:devDebugRuntimeClasspath' by consistent resolution
```

### 실패 이력
- 2026-04-18 09:09 (push) — failure
- 2026-04-20 08:20 (schedule) — failure
- 2026-04-21 06:39 / 06:51 (workflow_dispatch) — failure
- **2026-04-23 14:04 (workflow_dispatch)** — failure (PR #1682 env fix 적용 후 첫 실행)

**총 5회 연속 failure**. PR #1674 (patrol_test/ 디렉토리), PR #1682 (env file) 수정 후에도 여전히 빌드 단계에서 막힘.

## 원인

\`apps/app_user/android/app/build.gradle.kts\` 에 **서로 호환되지 않는** AndroidX test 의존성 지정:

```kotlin
androidTestImplementation("androidx.test:runner:1.6.2")    // ← 명시적 요구
androidTestUtil("androidx.test:orchestrator:1.5.1")        // ← 1.5.x 계열
```

Gradle 의 **consistent resolution** 기능이:
1. Runtime classpath (\`:app:devDebugRuntimeClasspath\`) 에서 orchestrator 1.5.1 이 끌어오는 \`runner:1.5.1\` 을 확정
2. AndroidTest classpath 도 **\`strictly 1.5.1\`** 로 잠금
3. \`androidTestImplementation\` 의 1.6.2 요구와 충돌 → resolution 실패

### AndroidX test 호환성 매트릭스 (참고)
- orchestrator \`1.5.1\` ↔ runner \`1.5.x\` (호환 쌍)
- orchestrator \`1.6.x\` ↔ runner \`1.6.x\` (호환 쌍)
- **혼용 불가**

## 해결 방법 (택일)

### 옵션 A — runner 1.5.1 로 downgrade (가장 간단, 낮은 리스크)

\`apps/app_user/android/app/build.gradle.kts\`:
```kotlin
androidTestImplementation(\"androidx.test:runner:1.5.1\")  // 1.6.2 → 1.5.1
androidTestUtil(\"androidx.test:orchestrator:1.5.1\")       // 그대로
```

- Orchestrator 와 버전 정렬 → consistent resolution 통과
- Patrol 4.5.0 이 runner 1.5.x 호환 여부 확인 필요 (Patrol 공식 docs)

### 옵션 B — orchestrator 1.6.2 로 upgrade

```kotlin
androidTestImplementation(\"androidx.test:runner:1.6.2\")
androidTestUtil(\"androidx.test:orchestrator:1.6.2\")  // 1.5.1 → 1.6.2
```

- 최신 버전 정렬
- AGP / compileSdk 호환 확인 필요 (1.6.x 는 SDK 34+ 권장)

### 옵션 C — resolutionStrategy.force (땜질, 비권장)

```kotlin
configurations.all {
  resolutionStrategy {
    force(\"androidx.test:runner:1.6.2\")
  }
}
```

- 다른 의존성이 runner 를 요구하는 패턴에 영향 가능
- 일시 방편. 근본 해결 아님

## 권장

**옵션 A (runner 1.5.1 downgrade)** 로 시작. 이유:
- 변경량 한 줄
- orchestrator 1.5.1 은 기존 코드가 검증한 조합
- 실패 시 옵션 B 로 롤백 가능

Patrol 4.5.0 문서에서 호환성 확인 후 PR.

## Partner App 도 동일?

\`apps/app_partner/android/app/build.gradle.kts\` 도 동일 패턴 있을 가능성. Worker 가 확인 후 같이 수정 필요.

## 검증

### PR 머지 후
- [ ] \`gh workflow run patrol-e2e.yml --ref dev\` 수동 trigger
- [ ] Gradle build 성공 (\`Failed to build apk\` 에러 없음)
- [ ] 최소 \`kakao_login_test\` / \`payment_pg_test\` / \`permission_grant_test\` 가 \`skip: true\` 상태여도 **실행 단계까지 도달** (skipped 로 종료 OK)
- [ ] Artifact upload step 도달 여부 확인

### 후속 (본 이슈 이후)
- Patrol 실제 테스트 pass 시나리오 작성 (현 3개 전부 TODO #1436 skip 처리 상태)
- #1778 CUJ → Patrol 이관 착수 가능 상태로 전환

## 스코프

### In scope
- \`apps/app_user/android/app/build.gradle.kts\` dependency 버전 정렬
- \`apps/app_partner/android/app/build.gradle.kts\` 동일 패턴 확인 + 수정

### Out of scope
- Patrol 테스트 실제 내용 작성 (skip 해제) — #1436 / #1778
- Emulator avd 설정 최적화
- CUJ 이관 (#1778)

## 관련

- **#1778** CUJ → Patrol 이관 (본 이슈가 **선결 blocker**)
- #1674 patrol_test/ 디렉토리 신설 (PR 머지됨)
- **#1682** patrol-e2e.yml env file fix (PR 머지됨)
- #1436 Patrol 테스트 실제 자격 확보 (미해결, skip 상태 유지 중)
- 실패 run 로그: run 24708423410, 24707973950, 24656060611, 24601458581, **24839817598**

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -375,8 +375,35 @@ jobs:
           fi
           echo "✅ All workflow YAML files are valid"
 
+  # Fix #1780: androidx.test runner/orchestrator 버전 정합성 검증
+  # runner와 orchestrator는 동일 major.minor 계열이어야 함 (1.5.x↔1.5.x, 1.6.x↔1.6.x)
+  # Gradle consistent resolution이 혼용 시 빌드 실패를 발생시키므로 PR 단계에서 사전 차단.
+  check-android-test-deps:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+      - name: Verify androidx.test runner/orchestrator version alignment
+        run: |
+          GRADLE_FILE="apps/app_user/android/app/build.gradle.kts"
+          RUNNER=$(grep 'androidx\.test:runner:' "$GRADLE_FILE" | grep -oP '\d+\.\d+\.\d+' | head -1)
+          ORCHESTRATOR=$(grep 'androidx\.test:orchestrator:' "$GRADLE_FILE" | grep -oP '\d+\.\d+\.\d+' | head -1)
+          echo "runner: $RUNNER"
+          echo "orchestrator: $ORCHESTRATOR"
+          if [ -z "$RUNNER" ] || [ -z "$ORCHESTRATOR" ]; then
+            echo "❌ Could not parse runner or orchestrator version from $GRADLE_FILE"
+            exit 1
+          fi
+          RUNNER_MINOR=$(echo "$RUNNER" | cut -d. -f1,2)
+          ORCHESTRATOR_MINOR=$(echo "$ORCHESTRATOR" | cut -d. -f1,2)
+          if [ "$RUNNER_MINOR" != "$ORCHESTRATOR_MINOR" ]; then
+            echo "❌ Version mismatch: runner $RUNNER vs orchestrator $ORCHESTRATOR"
+            echo "Both must share the same major.minor series (e.g., both 1.5.x or both 1.6.x)"
+            exit 1
+          fi
+          echo "✅ runner $RUNNER and orchestrator $ORCHESTRATOR are compatible ($RUNNER_MINOR series)"
+
   ci-result:
-    needs: [check-migration-versions, check-env-manifest-sync, check-workflow-yaml, test-flutter-apps, lint-landing-user, lint-landing-partner, test-supabase, test-edge-functions]
+    needs: [check-migration-versions, check-env-manifest-sync, check-workflow-yaml, check-android-test-deps, test-flutter-apps, lint-landing-user, lint-landing-partner, test-supabase, test-edge-functions]
     if: always()
     runs-on: ubuntu-latest
     steps:

--- a/apps/app_user/android/app/build.gradle.kts
+++ b/apps/app_user/android/app/build.gradle.kts
@@ -77,7 +77,8 @@ android {
 
 dependencies {
     coreLibraryDesugaring("com.android.tools:desugar_jdk_libs:2.1.4")
-    androidTestImplementation("androidx.test:runner:1.6.2")
+    // Fix #1780: runner 1.6.2 vs orchestrator 1.5.1 버전 충돌 — 같은 1.5.x 계열로 정렬
+    androidTestImplementation("androidx.test:runner:1.5.1")
     androidTestUtil("androidx.test:orchestrator:1.5.1")
 }
 

